src/platform/unix/linux/mod.rs

```diff
@@ -953,21 +976,32 @@ fn is_socket(fd: c_int) -> bool {
 
 // FFI stuff follows:
 
-const SCM_RIGHTS: c_int = 0x01;
+#[allow(non_snake_case)]
+fn CMSG_LEN(length: SockLen) -> SockLen {
+    (CMSG_ALIGN(mem::size_of::<cmsghdr>()) + length) as SockLen
+}
+
+#[allow(non_snake_case)]
+fn CMSG_ALIGN(length: size_t) -> SockLen {
+    ((length + mem::size_of::<size_t>() - 1) & !(mem::size_of::<size_t>() - 1)) as SockLen
+}
 
 #[allow(non_snake_case)]
-fn CMSG_LEN(length: size_t) -> size_t {
-    CMSG_ALIGN(mem::size_of::<cmsghdr>()) + length
+#[cfg(target_os = "linux")]
+unsafe fn CMSG_DATA(cmsg: *mut cmsghdr) -> *mut c_void {
+    cmsg.offset(1) as *mut c_void
```

Comment:
Didn't think about that. That would make this a bit more simplistic. I'd definitely be okay with making this change